Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/person/core/PersonCertificatesExtensions.kt

```diff
@@ -85,20 +85,31 @@ private fun Collection<CwaCovidCertificate>.rule2FindRecentRaCertificate(
  */
 private fun Collection<CwaCovidCertificate>.rule3FindRecentLastShot(
     nowUtc: Instant
-): CwaCovidCertificate? = this
-    .filterIsInstance<VaccinationCertificate>()
-    .filter {
-        with(it.rawCertificate.vaccination) { doseNumber == totalSeriesOfDoses }
-    }
-    .filter {
-        Days.daysBetween(it.rawCertificate.vaccination.vaccinatedOn, nowUtc.toLocalDateUtc()).days > 14
-    }
-    .maxWithOrNull(
-        compareBy(
-            { it.rawCertificate.vaccination.vaccinatedOn },
-            { it.headerIssuedAt }
+): CwaCovidCertificate? {
+    val vaccinesWithTwoDoses = listOf("EU/1/20/1528", "EU/1/21/1529", "EU/1/20/1507")
```

Comment:
If this really needs to change, we should look for a more sustainable solution. We could just sort for dose number, so 3/3 would be selected over 2/2

Reply:
What I'm trying to do here is differentiate between vaccines that require two doses to get full immunity and the ones that only require one dose like J&J. In the case of J&J, a vaccine whith dose 2/2 would already be considered a booster, right?